src/Twitter.php

```diff
@@ -325,16 +334,19 @@ private function formatResponse(Response $response, string $format)
      * @param array  $parameters
      * @param string $method
      *
-     * @throws GuzzleException
      * @return mixed|string
      */
     private function request(string $url, array $parameters, string $method)
     {
-        $format = $parameters[self::KEY_FORMAT] ?? self::RESPONSE_FORMAT_OBJECT;
-        $requestOptions = $this->getRequestOptions($parameters, $method);
+        $responseFormat = $parameters[self::KEY_RESPONSE_FORMAT] ?? self::RESPONSE_FORMAT_OBJECT;
+        $requestFormat = $parameters[self::KEY_REQUEST_FORMAT] ?? $responseFormat;
+
+        unset($parameters[self::KEY_RESPONSE_FORMAT]);
```

Comment:
Do we need to unset the request format here?

Reply:
Not really but I did it so that we make no changes to the `$parameters` inside `getRequestOptions()`.